livekit-agents/livekit/agents/ipc/job_context.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 LiveKit, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Coroutine, Optional, TYPE_CHECKING
+from livekit import api, rtc
+from livekit.protocol import worker, agent
+
+import asyncio
+import contextlib
+import logging
+from .client import IPCClient
+
+
+class JobContext:
```

Comment:
nit: it's a bit strange why the context is located in the ipc package. two reasons:
* the context seems like a piece of code the end user would have to access frequently (I know I've had to reference it a few times when creating an agent)
* while it does communicate with parent process, it feels like that's not the core of this class